src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/ProvenBlockHeaderStore.cs

```diff
@@ -135,9 +135,31 @@ public async Task<ChainedHeader> InitializeAsync(ChainedHeader highestHeader)
             {
                 // Repository is behind chain of headers.
                 tip = tip.FindAncestorOrSelf(repoTip.Hash, repoTip.Height);
+
+                if (tip == null)
+                {
+                    // Start at one less of the current repo height as we have already checked
+                    // the repo tip.
+                    var height = repoTip.Height - 1;
+                    while (true)
```

Comment:
why not `for` loop? 
In case there is no header with hash that is equal to heighestHeader.Hash it will be an infinite loop.

I think it's better to iterate till genesis and if it wasn't found- throw an exception.

Reply:
Ah yes of course! So if its `null` throw an exception?